fetch.js

```diff
@@ -378,4 +378,4 @@
     })
   }
   self.fetch.polyfill = true
-})();
+})(this.self || this);
```

Comment:
I feel like I would like to revert to @chrisirhc's idea by doing

```js
typeof self !== 'undefined' ? self : this
```

simply because that approach has been better battle-tested by compilers such as Browserify.